scijava/scijava-struct/src/main/java/org/scijava/struct/ItemIO.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * #%L
+ * SciJava Common shared library for SciJava software.
+ * %%
+ * Copyright (C) 2009 - 2017 Board of Regents of the University of
+ * Wisconsin-Madison, Broad Institute of MIT and Harvard, Max Planck
+ * Institute of Molecular Cell Biology and Genetics, University of
+ * Konstanz, and KNIME GmbH.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.struct;
+
+/**
+ * Defines the input/output type of a struct {@link Member}.
+ *
+ * @author Curtis Rueden
+ * @author David Kolb
+ */
+public enum ItemIO {
+	/**
+	 * The item is an input for the operation.
+	 */
+	INPUT, 
+	/**
+	 * The item is an output for the operation.
+	 */
+	OUTPUT,
+	/**
+	 * The item is a hybrid input/output for the operation, whose structure is fixed
+	 * but whose content will be populated during execution of the operation.
+	 * 
+	 * @see org.scijava.param.Container
+	 * @see org.scijava.ops.function.Computers
+	 */
+	CONTAINER,
+	/**
+	 * The item is a hybrid input/output for the operation, whose current value will
+	 * be used as input to the computation, but subsequently overwritten with a
+	 * result.
+	 * 
+	 * @see org.scijava.param.Mutable
+	 * @see org.scijava.ops.function.Inplaces
```

Comment:
struct does not depend on any scijava ops components, right? So this `@see` is not valid. It's great to point out that `MUTABLE` is relevant to inplace functions in ops, but I think we need to do it in a "non-type-safe" way, unfortunately.